widgets/more_buttons.py

```diff
@@ -0,0 +1,40 @@
+from PyQt5.QtWidgets import QWidget, QVBoxLayout, QPushButton
+from lib.libbol import *
+from widgets.tree_view import BolHeader
+
+class MoreButtons(QWidget):
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.parent = parent
+        self.vbox = QVBoxLayout(self)
+
+    def add_buttons(self, option = None):
+        self.clear_buttons()
+
+        if option is None or isinstance(option, BolHeader):
+            return
+
+        obj = option.bound_to
+        print(obj)
+        if isinstance(obj, EnemyPointGroups):
+            new_enemy_group = QPushButton(self)
+            new_enemy_group.setText("Add Enemy Group")
+            new_enemy_group.clicked.connect( \
+                lambda: self.parent.button_side_button_action("add_enemygroup", obj) )
+            self.vbox.addWidget(new_enemy_group)
+        elif isinstance(obj, EnemyPointGroup):
+            new_enemy_point = QPushButton(self)
+            new_enemy_point.setText("Add Enemy Points To End")
```

Comment:
Probably very opinionated, but I _think_ there should be only two possible buttons:
- **Add Enemy Path**
- **Insert Enemy Points**

As mentioned in another comment, we should use the index of the currently select item in the tree as the insertion point (without needing the `points_added` member that is all over the place at the moment).

The logic as I see it would be:

- If the top-level **Enemy Paths** tree item is selected, then the **Add Enemy Path** button is shown.
- If **Add Enemy Path** is clicked, a new enemy path is appended in a selected state, and is [made visible](https://doc.qt.io/qt-6/qtreewidget.html#scrollToItem) in the tree.
- If an enemy path or an enemy point is selected, then the **Insert Enemy Points** button is shown.
- If **Insert Enemy Points** is clicked, it activates insertion mode.
- If the user clicks on the viewport while in insertion mode:
  - If an enemy path is selected, a new enemy point is inserted at index 0 in a selected state, and is made visible in the tree.
  - If an enemy point is selected, a new enemy point is inserted at the current position + 1 in a selected state, and is made visible in the tree.
- If at any point the user selects something that is not an enemy path or enemy point, insertion mode is dismissed.

Thoughts?

Reply:
An enemy path being selected upon creation and scrollToItem sounds like a good idea. 

> If an enemy path is selected, a new enemy point is inserted at index 0 in a selected state, and is made visible in the tree.

Why index 0? I would honestly be in favor of not including any options for individual enemy paths at all, as there is no one logical place to insert points.

Reply:
This is to support _prepending_. Otherwise there is no way to insert enemy points at index 0.

Reply:
I had not considered that. I can change the functionality, but I don't think that "Insert Enemy Points" adequately conveys that points will be added to the start.

Reply:
Insert is the general term. We could have different labels for each case (Prepend, Insert, Append), but I'd be happy with just the general term.

Reply:
Changed the button to add to the start instead.

Reply:
This was further discussed offline, and was addressed!